components/debounce/browser/debounce_navigation_throttle.cc

```diff
@@ -90,6 +93,17 @@ DebounceNavigationThrottle::MaybeCreateThrottleFor(
   if (!debounce_service)
     return nullptr;
 
+  WebContents* web_contents = navigation_handle->GetWebContents();
```

Comment:
this would be a lot simpler if you added `IsDebounceEnabled` to `DebounceService` since it already has the prefs